### PR TITLE
docs: add community documentation

### PR DESCRIPTION
**Pull request by mgechev**
This PR introduces:

1. Issue templates for bug reports and feature requests
1. Pull request template
1. Code of conduct, which reuses the content from angular/angular
1. Guidelines for contributing to the project
1. A development document which explains how to get started with fixing
issues/implementing new rules. There's more work here, this is just the
initial draft
1. Update of the readme to introduce project scope

### CODE REVIEW

**Comment by alexeagle** [comment]
Hmm, maybe the `.github` folder should not be synced internally at google3? The plus side is you can merge those changes directly on github. The minus is that you couldn't change those files atomically in the same commit with files that are synced.

**Comment by mgechev** [comment]
I'd vote for syncing everything to google3 so that we follow the same process for all PRs.

File: DEVELOPING.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# For Developers
+
+We strongly encourage you to review the project's scope described in the `README.md` file before working on new features. For large changes, consider writing a design document using [this template](https://docs.google.com/document/d/1U6GlAaPF3tl1wvBNXebW1mD3oHZxV5h9TH-bR0p8Uz8/edit?usp=sharing).
```
Comment (gregmagolan) [inline DEVELOPING.md]: Document is not public (pointed out by @filipesilva in #348)

Comment (mgechev) [reply DEVELOPING.md]: Thanks for pointing this out! Will sync today.

**Pull request closed without merging**